decryptor/postgresql/data_encoder.go

```diff
@@ -0,0 +1,255 @@
+)
+
+// PgSQLDataEncoderProcessor implements processor and encode/decode binary intX values to text format which acceptable by Tokenizer
+type PgSQLDataEncoderProcessor struct {
+	mode DataEncoderMode
+}
+
+// NewPgSQLDataEncoderProcessor return new data encoder/decoder from/to binary format for tokenization
+func NewPgSQLDataEncoderProcessor(mode DataEncoderMode) (*PgSQLDataEncoderProcessor, error) {
+	switch mode {
+	case DataEncoderModeDecode, DataEncoderModeEncode:
+		return &PgSQLDataEncoderProcessor{mode}, nil
+	}
+	return nil, ErrInvalidDataEncoderMode
+}
+
+// ID return name of processor
+func (p *PgSQLDataEncoderProcessor) ID() string {
+	return "PgSQLDataEncoderProcessor"
+}
+
+// here we process encryption/tokenization results before send it to a client
+// acra decrypts or de-tokenize SQL literals, so we should convert string SQL literals to binary format
+// if client expects int, then parse INT literals and convert to binary 4/8 byte format
+// if expects bytes, then pass as is
+// if expects string, then leave as is if it is valid string or encode to hex
+// if it is encrypted data then we return default values or as is if applicable (binary data)
+func (p *PgSQLDataEncoderProcessor) encodeBinary(ctx context.Context, data []byte, setting config.ColumnEncryptionSetting, columnInfo base.ColumnInfo, logger *logrus.Entry) (context.Context, []byte, error) {
+	if len(data) == 0 {
+		return ctx, data, nil
+	}
+	switch setting.GetEncryptedDataType() {
+	case common2.EncryptedType_String:
+		if !base.IsDecryptedFromContext(ctx) {
+			value := setting.GetDefaultDataValue()
+			if value != nil {
+				return ctx, []byte(*value), nil
+			}
+		}
+		return ctx, data, nil
+	case common2.EncryptedType_Bytes:
+		if !base.IsDecryptedFromContext(ctx) {
+			value := setting.GetDefaultDataValue()
+			if value != nil {
+				binValue, err := base64.StdEncoding.DecodeString(*value)
+				if err == nil {
+					return ctx, binValue, nil
+				}
+				logger.WithError(err).Errorln("Can't decode base64 default value")
+			}
+		}
+		return ctx, data, nil
+	case common2.EncryptedType_Int32, common2.EncryptedType_Int64:
+		size := 8
+		if setting.GetEncryptedDataType() == common2.EncryptedType_Int32 {
+			size = 4
+		}
+		// convert back from text to binary
+		value, err := strconv.ParseInt(string(data), 10, 64)
+		// we don't return error to not cause connection drop on Acra side and pass it to app to deal with it
+		if err != nil {
+			if setting.GetDefaultDataValue() != nil {
+				newVal := setting.GetDefaultDataValue()
+				value, err = strconv.ParseInt(*newVal, 10, 64)
+				if err != nil {
+					logger.WithError(err).Errorln("Can't parse default integer value")
+					return ctx, data, err
+				}
+			} else {
+				return ctx, data, nil
+			}
+		}
+		newData := make([]byte, size)
+		switch size {
+		case 4:
+			binary.BigEndian.PutUint32(newData, uint32(value))
+			break
+		case 8:
+			binary.BigEndian.PutUint64(newData, uint64(value))
+			break
+		default:
```

Comment:
I think, there is no need of default block because size is our local variable that could be only 4 or 8, right?